src/Host/Protocol/Test/RHostPipe/RHostPipeTest.cs

```diff
@@ -0,0 +1,193 @@
+            await stream.WriteAsync(input, 0, input.Length, ct);
+        }
+
+        private async Task ReadAnyOutputFromRHostAsync(Stream stream, CancellationToken ct) {
+            // Capture R messages after sending fuzzed input.
+            byte[] buffer = new byte[1024 * 1024];
+            int bytesRead = 0;
+            do {
+                bytesRead = await stream.ReadAsync(buffer, 0, buffer.Length, ct);
+            } while (!ct.IsCancellationRequested);
+        }
+
+        private Interpreter _interpreter;
+
+        [Test]
+        [Category.FuzzTest]
+        public async Task RHostPipeFuzzTest() {
+            var interpreters = GetInterpreters();
+            _interpreter = interpreters.FirstOrDefault();
+
+            for (int i = 0; i < 100; ++i) {
+                byte[] input = GenerateInput();
+                await RHostPipeFuzzTestRunnerAsync(input);
+            }
+        }
+
+        private byte[] GenerateInput() {
+            using (MemoryStream ms = new MemoryStream())
+            using (BinaryWriter writer = new BinaryWriter(ms)) {
+                // Id =uint64
+                writer.Write(GenerateUInt64());
+                // Request id = uint64
+                writer.Write(GenerateUInt64());
+                // Name
+                writer.Write(GenerateAsciiString().ToCharArray());
+                writer.Write('\0');
+                // Json ARRAY string = {[, , , ]} 
+                writer.Write(GenerateJsonArray().ToCharArray());
+                writer.Write('\0');
+                // blob
+                writer.Write(GenerateBytes());
+                writer.Flush();
+                return ms.ToArray();
+            }
+        }
+
+        private string GenerateAsciiString(int max = 10000) {
+            byte[] rbuf = new byte[_rand.Next(0, max)];
+            _rand.NextBytes(rbuf);
+            return Encoding.ASCII.GetString(rbuf);
+        }
+
+        private string GenerateUTF8String(int max = 10000) {
+            byte[] rbuf = new byte[_rand.Next(0, max)];
+            _rand.NextBytes(rbuf);
+            return Encoding.UTF8.GetString(rbuf);
+        }
+
+        private string GenerateJsonArray() {
+            if(_rand.Next()%2 == 0) {
+                // return random unicode string
+                return GenerateUTF8String();
+            }
+
+            string[] parts = new string[_rand.Next(0,1000)]; 
+            for(int i = 0; i < parts.Length; ++i) {
+                parts[i] = GenerateUTF8String(100);
+            }
+
+            // return a JSON array of random unicode strings
+            return "{[" + string.Join(",", parts) + "]}";
+        }
+
+        private byte[] GenerateBytes(int max = 1000000) {
+            byte[] rbuf = new byte[_rand.Next(0, max)];
+            _rand.NextBytes(rbuf);
+            return rbuf;
+        }
+
+        private ulong GenerateUInt64() {
```

Comment:
These methods can be useful in other tests. You may want to add `Random` class to the `Microsoft.UnitTests.Core` assembly.